lib/web_ui/lib/src/engine/compositor/skia_object_cache.dart

```diff
@@ -0,0 +1,238 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+part of engine;
+
+/// A cache of Skia objects whose memory Flutter manages.
+///
+/// When using Skia, Flutter creates Skia objects which are allocated in
+/// WASM memory and which must be explicitly deleted. In the case of Flutter
+/// mobile, the Skia objects are wrapped by a C++ class which is destroyed
+/// when the associated Dart object is garbage collected.
+///
+/// On the web, we cannot tell when a Dart object is garbage collected, so
+/// we must use other strategies to know when to delete a Skia object. Some
+/// objects, like [ui.Paint], can safely delete their associated Skia object
+/// because they can always recreate the Skia object from data stored in the
+/// Dart object. Other objects, like [ui.Picture], can be serialized to a
+/// JS-managed data structure when they are deleted so that when the associated
+/// object is garbage collected, so is the serialized data.
+class SkiaObjectCache {
+  final int maximumSize;
+
+  final DoubleLinkedQueue<SkiaObject> _itemQueue;
+  final Map<SkiaObject, DoubleLinkedQueueEntry<SkiaObject>> _itemMap;
```

Comment:
Doc comments on these fields explaining what they contain would be useful.

Reply:
Done.